src/libpmempool/replica.c

```diff
@@ -223,6 +306,34 @@ replica_is_part_broken(unsigned repn, unsigned partn,
 		(PART_HEALTH(rhs, partn) & IS_BROKEN);
 }
 
+/*
+ * replica_mark_no_badblocks -- mark replica as not having bad blocks
+ */
+void
+replica_mark_no_badblocks(unsigned repn, struct poolset_health_status *set_hs)
```

Comment:
not sure if this makes sense - should be also mark all parts of this replica as having no badblocks? Then we could potentially simplify the logic where we remove the HAS_BAD_BLOCKS flag from the replica and the `replica_has_bad_blocks` just goes over all parts looking for badblocks.

It might be pointless, but just something I noticed - feel free to ignore.

Reply:
I moved ```replica_mark_no_badblocks``` and ```replica_mark_part_no_badblocks``` to sync.c and made them internal. They are only a "wrappers" for some ugly lines of code.